projects/libyaml/libyaml_dumper_fuzzer.c

```diff
@@ -0,0 +1,314 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "yaml.h"
+#include "yaml_write_handler.h"
+#include <assert.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#ifdef NDEBUG
+#undef NDEBUG
+#endif
+
+#define MAX_DOCUMENTS 16
+
+bool nodes_equal(yaml_document_t *document1, int index1,
+                 yaml_document_t *document2, int index2, int level) {
+  bool equal = true;
+
+  if (level++ > 1000)
+    return !equal;
+  yaml_node_t *node1 = yaml_document_get_node(document1, index1);
+
+  if (!node1)
+    return !equal;
+
+  yaml_node_t *node2 = yaml_document_get_node(document2, index2);
+
+  if (!node2)
+    return !equal;
+
+  int k;
```

Comment:
is this unused? (it seems to me it is)

Reply:
actually there are some unused "int k" 's scattered around, remove the unused ones